src/axom/slic/tests/slic_log_stream_status_monitor.cpp

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/lumberjack/Lumberjack.hpp"
+#include "axom/lumberjack/RootCommunicator.hpp"
+#include "axom/slic/core/LogStreamStatusMonitor.hpp"
+#include "axom/slic/streams/GenericOutputStream.hpp"
+#include "axom/slic/streams/LumberjackStream.hpp"
+#include "axom/slic/streams/SynchronizedStream.hpp"
+#include "gtest/gtest.h"
+
+#include <mpi.h>
+
+enum class StreamType
+{
+    Generic,
+    Synchronized,
+    Lumberjack
+};
+
+class LogStreamStatusMonitorParamTest : public ::testing::TestWithParam<StreamType>
+{
+protected:
+    std::ostringstream test_stream;
+    std::unique_ptr<axom::slic::LogStream> stream;
+    axom::slic::LogStreamStatusMonitor logStreamStatusMonitor;
+
+    void SetUp() override
+    {
+      switch(GetParam())
+      {
+        case StreamType::Generic:
+          stream = std::make_unique<axom::slic::GenericOutputStream>(&test_stream);
+          break;
+        case StreamType::Synchronized:
+          stream = std::make_unique<axom::slic::SynchronizedStream>(&test_stream, MPI_COMM_WORLD);
+          break;
+        case StreamType::Lumberjack:
+          stream = std::make_unique<axom::slic::LumberjackStream>(&test_stream, MPI_COMM_WORLD, 1);
+          break;
+      }
+    }
+};
+
+// namespace alias
+namespace slic = axom::slic;
+
+TEST_P(LogStreamStatusMonitorParamTest, test_has_pending_messages)
+{
+  logStreamStatusMonitor.addStream(stream.get());
+
+  EXPECT_EQ(logStreamStatusMonitor.hasPendingMessages(), false);
+
+  stream->append(axom::slic::message::Debug, "test message", "test tag", "test file name", 1, false, false);
+
+  if(stream->canTestHasPendingMessages() == true)
+  {
+    EXPECT_EQ(logStreamStatusMonitor.hasPendingMessages(), true);
+  }
+  else
+  {
+    EXPECT_EQ(logStreamStatusMonitor.hasPendingMessages(), false);
+  }
```

Comment:
Thanks for updating the tests -- this is a lot cleaner!

One more suggestion:
```suggestion
  if(stream->canTestHasPendingMessages() == true)
  {
    EXPECT_TRUE(logStreamStatusMonitor.hasPendingMessages());
  }
  else
  {
    EXPECT_FALSE(logStreamStatusMonitor.hasPendingMessages());
  }
```